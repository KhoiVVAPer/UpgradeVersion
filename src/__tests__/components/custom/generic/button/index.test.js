import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../../../../../components/custom/generic/button';

const setUp = (props) => {
  const component = shallow(<Button {...props} />);
  return component;
};


describe('components > custom > elements > Button', () => {
  let component;
  let props = {
    text: 'Button',
  };
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders TouchableDebounce component', () => {
    const input = component.find('TouchableDebounce').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders Text component', () => {
    const input = component.find('Text').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders Loading image component when loading prop is true', () => {
    props = {
      ...props,
      loading: true
    };
    component = setUp(props);
    const input = component.find('Image').length;
    const output = 1;
    expect(input).toEqual(output);
  });
});
