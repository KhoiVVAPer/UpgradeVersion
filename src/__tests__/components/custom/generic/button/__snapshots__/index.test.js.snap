// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > Button Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    Button
  </react-native-mock>
</react-native-mock>
`;
